Added documentation for grok_when option (#5912)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/processors/grok.md

> # grok

Reviewer: Should this be capitalized if the name of the processor is `Grok`?

> The `Grok` processor takes unstructured data and utilizes pattern matching to structure and extract important keys.

Reviewer: ```suggestion
The `Grok` processor uses pattern matching to structure and extract important keys from unstructured data.
```

## Configuration

The following table describes options you can use with the `Grok` processor to structure your data and make your data easier to query.

Option | Required | Type | Description
:--- | :--- | :--- | :---
break_on_match | No | Boolean | Specifies whether to match all patterns or stop once the first successful match is found. Default value is `true`.

> grok_when | No | String | Specifies the condition when the grok processor should do matching. Default is no condition.

Reviewer: ```suggestion
grok_when | No | String | Specifies under what condition the `Grok` processor should perform matching. Default is no condition.
```

> keep_empty_captures | No | Boolean | Enables preserving `null` captures. Default value is `false`.

Reviewer: ```suggestion
keep_empty_captures | No | Boolean | Enables the preservation of `null` captures. Default value is `false`.
```

Content will be added to this section.--->

> ## Conditional grok

Reviewer: Should "Grok" be capitalized?

> The grok processor can be configured to run conditionally by using `grok_when` option. An example grok processor config when `grok_when` -

Reviewer 2: ```suggestion
The grok processor can be configured to run conditionally by using the `grok_when` option. The following is an example grok processor configuration using `grok_when`:
```

> The grok processor can be configured to run conditionally by using the `grok_when` option. The following is an example grok processor configuration using `grok_when`:

Reviewer: ```suggestion
The `Grok` processor can be configured to run conditionally by using the `grok_when` option. The following is an example `Grok` processor configuration that uses `grok_when`:
```

```
processor:
  - grok:
      grok_when: '/type == "ipv4"'
        match:
          message: ['%{IPV4:clientip} %{WORD:request} %{POSINT:bytes}']
  - grok:
      grok_when: '/type == "ipv6"'
        match:
          message: ['%{IPV6:clientip} %{WORD:request} %{POSINT:bytes}']
```

> `grok_when` option takes a conditional expression which described in detail [here](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/)

Reviewer 2: ```suggestion
The `grok_when` option takes a conditional expression. This expression is detailed in the [Expression syntax](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/) documentation.
```

> The `grok_when` option takes a conditional expression. This expression is detailed in the [Expression syntax](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/) documentation.

Reviewer: ```suggestion
The `grok_when` option can take a conditional expression. This expression is detailed in the [Expression syntax](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/) documentation.
```

### Timer

> * `grokProcessingTime`: The time taken by individual records to match against patterns from `match`. The `avg` metric is the most useful metric for this timer because it provides the average value of the time it takes records to match.

Reviewer: ```suggestion
* `grokProcessingTime`: The time taken by individual records to match against `match` patterns. The `avg` metric is the most useful metric for this timer because it provides the average value of the time it takes records to match.
```

> * `grokProcessingTime`: The time taken by individual records to match against patterns from `match`. The `avg` metric is the most useful metric for this timer because it provides the average value of the time it takes records to match.

Reviewer: "because it provides the average time taken to match records"?

> * `grokProcessingTime`: The time taken by individual records to match against patterns from `match`. The `avg` metric is the most useful metric for this timer because it provides you with an average value of the time it takes records to match.

Reviewer 2: ```suggestion
* `grokProcessingTime`: The time taken by individual records to match against patterns from `match`. The `avg` metric is the most useful metric for this timer because it provides the average value of the time it takes records to match.
```
